service/lib/agama/dbus/storage/device.rb

```diff
@@ -77,14 +79,16 @@ def storage_device=(value)
         attr_reader :tree
 
         # Adds the required interfaces according to the storage object
-        def add_interfaces # rubocop:disable Metrics/CyclomaticComplexity
+        def add_interfaces # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
```

Comment:
well, this growing kind of call for a bit of refactoring. Something like having in interface class method like `apply?(storage_device)` and now have here something like
```
require "agama/dbus/storage/interfaces"

...
   interfaces = Interfaces.constants.select {|c| Interfaces.const_get(c).is_a?(Class)}
   interfaces.select! { |i| i.apply?(storage_device) }
```

This way it will automatically recognize new interfaces if they are added to that summary interfaces file. It also moves responsibility of knowing what is requirement of interface to that file and keep this file much simplier and without breaking metrics that already indicate that you are doing something wrong here.

Reply:
ah, interfaces are components and not classes so simply replace `Class` with `Module`

Reply:
I like it! Thanks for your suggestion.